thirdparties/glog/glog.patch

```diff
@@ -371,6 +371,15 @@ index 0c86cf6..b77a4d8 100644
    delete sinks_;
    sinks_ = NULL;
  }
+@@ -906,7 +984,7 @@ bool LogFileObject::CreateLogfile(const string& time_pid_string) {
+   string string_filename = base_filename_+filename_extension_+
+                            time_pid_string;
+   const char* filename = string_filename.c_str();
+-  int fd = open(filename, O_WRONLY | O_CREAT | O_EXCL, FLAGS_logfile_mode);
++  int fd = open(filename, O_WRONLY | O_CREAT | O_APPEND | O_EXCL, FLAGS_logfile_mode);
```

Comment:
this line code https://github.com/google/glog/blob/503e3dec8d1fe071376befc62119a837c26612a3/src/logging.cc#L1123 may break your expectation that child process can write logs into the parent process's log file.

but, anyway, it's far more difficult to fix all cases, we can leave it behind, and fix this latter.

Reply:
> 

I had invoke `PidHasChanged()` before create log file, so the logger will never create a new file when pid changed.